project/dexter-core/src/test/com/samsung/sec/dexter/core/analyzer/AnalysisResultFileManagerTest.java

```diff
@@ -113,7 +113,37 @@ public void shouldWriteResultJsonFileWhenHasValidResultList() {
 		assertEquals(expectedMetricString, resultMap.get(ResultFileConstant.CODE_METRICS).toString());
 		
 		// check defect list
-		final String expectedDefectListString = "[{gdid=-1.0, occurences=[{code=, startLine=79.0, endLine=79.0, charStart=-1.0, charEnd=-1.0, variableName=, stringValue=, fieldName=, message=Memory leak: sgtree}], createdDateTime=1.439973272636E12, message=[#1@79] Memory leak: sgtree [#1@79] Memory leak: sgtree , severityCode=CRI, modifiedDateTime=1.439973272636E12, checkerCode=memleak, className=TestSource, methodName=testMethod, toolName=my-tool, language=JAVA, fileName=TestSource.java, modulePath=com/samsung}]";
+		final String expectedDefectListString = "[{gdid=-1.0, occurences=[{code=, startLine=79.0, endLine=79.0, charStart=-1.0, charEnd=-1.0, variableName=, stringValue=, fieldName=, message=Memory leak: sgtree}], createdDateTime=1.439973272636E12, message=[#1@79] Memory leak: sgtree [#1@79] Memory leak: sgtree , severityCode=CRI, categoryName=security, modifiedDateTime=1.439973272636E12, checkerCode=memleak, className=TestSource, methodName=testMethod, toolName=my-tool, language=JAVA, fileName=TestSource.java, modulePath=com/samsung, fileStatus=}]";
 		assertEquals(expectedDefectListString, resultMap.get(ResultFileConstant.DEFECT_LIST).toString());
 	}
+	
+	@Test
+	public void shouldReturnProperResultFilePrefixName(){
+		final String modulePath = "module_path";
+		final int modulePathHashcode = modulePath.hashCode();
+		final String fileName = "test.cpp";
+		
+		assertEquals(ResultFileConstant.RESULF_FILE_PREFIX + fileName + "_" + modulePathHashcode, 
+				AnalysisResultFileManager.getInstance().getResultFilePrefixName(modulePath, fileName));
+	}
+	
+	@Test
+	public void shouldReturnProperResultFilePrefixNameWithoutModulePath(){
+		final String modulePath = "";
+		final int modulePathHashcode = modulePath.hashCode();
+		final String fileName = "test.cpp";
+		
+		assertEquals(ResultFileConstant.RESULF_FILE_PREFIX + fileName + "_" + modulePathHashcode, 
+				AnalysisResultFileManager.getInstance().getResultFilePrefixName(modulePath, fileName));
+	}
+	
+	@Test
+	public void shouldReturnProperResultFilePrefixNameWithLongModulePath(){
+		final String modulePath = "abcdefghijklmnopqrstuvwxyz/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/1234567890/";
+		final int modulePathHashcode = modulePath.hashCode();
+		final String fileName = "test.cpp";
```

Comment:
It doesn't matter what a char type is.
refer to here : http://www.tutorialspoint.com/java/java_string_hashcode.htm
s[0]*31^(n-1) + s[1]*31^(n-2) + ... + s[n-1]